src/libs/Network/RetryCounter.js

```diff
@@ -0,0 +1,27 @@
+export default class RetryCounter {
+    constructor() {
+        this.retryMap = new Map();
+    }
+
+    clear() {
+        this.retryMap.clear();
+    }
+
+    /**
+     * @param {Object} request
+     * @returns {Number} retry count
+     */
+    incrementRetries(request) {
+        const current = this.retryMap.get(request) || 0;
```

Comment:
That makes sense to me, but at the same time I'm going to leave this since the code already existed and if it's confusing at least it's abstracted in this class for now :D